### PR TITLE
PXD-1869 Feat/dropdown

### PR DESCRIPTION
**Pull request by qingyashu**

### CODE REVIEW

**Review by MichaelLukowski — APPROVED** [review]
looks good

**Review by abgeorge7 — COMMENTED** [review]
A few comments. And then in the Storybook, when I click on the "separated" dropdown button only the arrow part of the button is highlighted orange - is that expected? I also think maybe the margin/padding is off on the more dropdown button between the text and the icon?

screen shot 2018-10-02 at 2 25 07 pm

File: src/components/Dropdown/Dropdown.css [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+.g3-dropdown__toggle--disabled .g3-dropdown__toggle-icon {
+  background-color: var(--g3-disabled-btn__color);
+}
+
+.g3-dropdown__toggle--disabled .g3-dropdown__toggle-separated {
+  border-left: none;
+}
+
+
+.g3-dropdown__toggle-icon--shown {
```
[the remark this answers is not in the document]
Comment (qingyashu) [reply src/components/Dropdown/Dropdown.css]: For menu opening status - I will change the name to `toggled`...  😅

File: src/components/Dropdown/Dropdown.css [hunk 2]
```diff
@@ -0,0 +1,227 @@
+@import "../../css/base.css";
```
[the remark this answers is not in the document]
Comment (qingyashu) [reply src/components/Dropdown/Dropdown.css:1]: Thanks~ will try to improve the name~

**Comment by qingyashu** [comment]
Thanks very much for the review comments!! I have update the code in:
1. Change some variable name and class name, especially in DropdownButtton files. Please have a look a again and see if they are clearer.
2. Fix indents, widths, border colors, and children props, also add more comments about props. 
More comments welcome~~~ 😊

File: src/components/Dropdown/Dropdown.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    </Dropdown>,
+  );
+  const separateWrapper = mount(
+    <Dropdown>
+      <Dropdown.Toggle separate onClick={func2}>
+        Dropdown
+      </Dropdown.Toggle>
+      <Dropdown.Menu>
+        <Dropdown.Item onClick={func3}>
+item1
```
[the remark this answers is not in the document]
Comment (abgeorge7) [reply src/components/Dropdown/Dropdown.test.js]: I think there's still weird indentation right?

File: src/components/Dropdown/DropdownButton.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import './DropdownButton.css';
+
+class DropdownButton extends Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
```
Comment (abgeorge7) [inline src/components/Dropdown/DropdownButton.jsx]: with arrow functions you don't have to do this :)

File: src/components/Dropdown/DropdownButton.jsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      <div
+        ref={this.props.menuTriggerElementRef}
+        className={`g3-dropdown-button__wrapper ${wrapperDisableStatusClassName} ${this.props.className || ''}`}
+      >
+
+        {/* Render dropdown button  */}
+        <button
+          type='button'
+          className={`g3-dropdown-button__button ${buttonIsSplitClassName} ${buttonTypeClassName}`}
+          onClick={this.props.split ? this.handleClick : this.props.handleTriggerMenu}
```
Comment (abgeorge7) [inline src/components/Dropdown/DropdownButton.jsx]: just wondering, do we need two separate props for "onClick" and "handleTriggerMenu"? if its a dropdown button shouldn't it open a menu either way, regardless of if its split or not?

File: src/components/Dropdown/Dropdown.test.js [hunk 6]
```diff
@@ -6,14 +6,15 @@ describe('<Dropdown />', () => {
...
+  /* eslint-disable */
   const dropdownWrapper = mount(
     <Dropdown>
       <Dropdown.Button>
         Dropdown
       </Dropdown.Button>
       <Dropdown.Menu>
         <Dropdown.Item onClick={func1}>
-item1
+          item1
```
Comment (abgeorge7) [inline src/components/Dropdown/Dropdown.test.js:17]: 🙌

**Review by abgeorge7 — APPROVED** [review]
🎉 looks good, the one thing is that it seems a bit odd to me that the non-menu trigger part of the split dropdown is clickable and turns orange on click because I don't think it does anything right? but the menu trigger part doesn't turn orange on click. but not a huge priority if you don't have time since we're not really using that type of dropdown. just make a ticket if you want to do it in a different PR :)

**Pull request merged**